### ISSUE TITLE
Feature request: Allow hover to show indent starting line to show the line before and the closing line

### ISSUE THREAD

**Issue by BillDenton** [SirTori/indenticator#4]
For example in C/C++:
for(i = 0; i < 10; i++)
{
   // stuff....
}
Hovering the near the indent line just shows the "{". This isn't much use. What is required is the line before. Obviously if the coding standard is different and uses:
for(i = 0; i < 10; i++) {
It produces useful information.
This would need to be a configurable option. I think may be a minimum length, so if the starting indent line was only one character it would display the line before. Alternatively a regular expression. A regular expression could cover the minimum length and allow matching to brackets etc. So this would probably be the best option (although more complex).
Also may display the closing line as well. For example:
typedef struct
{
   // Stuff...
} ATypeName;
In this case the opening "typedef struct {" isn't much use. The information is on the closing line. Similarly for do ... while loops.

**Comment by SirTori** [comment]
Thanks for the feedback.
I agree the current implementation isn't very useful in the described scenarios.

Your suggestions seem to be a sensible solution to this use case.
I'll think about the concrete implementation and configuration options I want to add and will release an update within in the week.

My Initial thoughts go to the following options and defaults:
```
/* Lines before the current indent to be shown on hover */
hover.peekBack: 1,
/* Lines after the current indent to be shown on hover */
hover.peekForward: 0,
/* Stop peeking if the peeked line has has different indent than first line around current indent block */
hover.stopPeekOnIndentChange: true, 
/* Remove lines from the hover at the beginning and end that have n characters or less */
hover.trimLinesShorterThan: 2,
/* Block placeholder to be written between peeked lines. Only inserted if peeked back and forward. */
hover.peekBlockPlaceholder: '...'
```

Example:
```
hover.PeekBack: 5,
hover.peekForward: 3,
hover.stopPeekOnIndentChange: true,
hover.trimLinesShorterThan: 2
hover.peekBlockPlaceholder: 'my awesome block'
```
on following snippet
```
  foo=bar;
}
do
{
  foo();
  bar();
} while(true);
foo();
bar();
foobar();
```

Would then result in a hover similar to:
```
do
{
  // my awesome block
} while(true);
foo();
bar();
```

Would this be a satisfactory solution for you?

**Comment by BillDenton** [comment]
Looks good. Thanks

**Comment by SirTori** [comment]
Just published Release 0.4.0
It includes more options to customize the hover content and even allows for language specific configuration. So you can make it work in a useful manner in all languages not just a single specific one.

See the [changelog](https://marketplace.visualstudio.com/items/SirTori.indenticator/changelog) for details and an example
